Source/WebCore/platform/spiel/PlatformSpeechSynthesizerSpiel.cpp

```diff
@@ -0,0 +1,258 @@
+
+    g_signal_connect_swapped(m_speaker.get(), "notify::paused", G_CALLBACK(+[](SpielSpeechWrapper* self, SpielSpeaker* speaker) {
+        gboolean isPaused;
+        g_object_get(speaker, "paused", &isPaused, nullptr);
+        if (isPaused)
+            self->m_platformSynthesizer.client().didPauseSpeaking(*self->m_utterance);
+        else
+            self->m_platformSynthesizer.client().didResumeSpeaking(*self->m_utterance);
+    }), this);
+
+    g_signal_connect_swapped(m_speaker.get(), "notify::voices", G_CALLBACK(+[](SpielSpeechWrapper* self, SpielSpeaker*) {
+        self->m_platformSynthesizer.client().voicesDidChange();
+    }), this);
+}
+
+SpielSpeechWrapper::~SpielSpeechWrapper()
+{
+    g_signal_handlers_disconnect_by_data(m_speaker.get(), this);
+}
+
+Vector<RefPtr<PlatformSpeechSynthesisVoice>> SpielSpeechWrapper::initializeVoiceList()
+{
+    Vector<RefPtr<PlatformSpeechSynthesisVoice>> platformVoices;
+    auto voices = spiel_speaker_get_voices(m_speaker.get());
+    unsigned position = 0;
+    m_voices.clear();
+    while (auto item = g_list_model_get_item(voices, position)) {
+        auto voice = SPIEL_VOICE(item);
+        const char* const* languages = spiel_voice_get_languages(voice);
+        if (!G_N_ELEMENTS(languages))
+            continue;
+
+        auto language = makeString(span(languages[0]));
+        auto name = makeString(span(spiel_voice_get_name(voice)));
+        auto identifier = makeString(span(spiel_voice_get_identifier(voice)));
+        bool isDefault = !position;
+        platformVoices.append(PlatformSpeechSynthesisVoice::create(identifier, name, language, false, isDefault));
+        m_voices.append(GRefPtr(voice));
+        position++;
+    }
+    return platformVoices;
+}
+
+void SpielSpeechWrapper::pause()
+{
+    if (!m_utterance)
+        return;
+
+    spiel_speaker_pause(m_speaker.get());
+}
+
+void SpielSpeechWrapper::resume()
+{
+    if (!m_utterance)
+        return;
+
+    spiel_speaker_resume(m_speaker.get());
+}
+
+void SpielSpeechWrapper::speakUtterance(RefPtr<PlatformSpeechSynthesisUtterance>&& utterance)
+{
+    ASSERT(!m_utterance);
+    ASSERT(utterance);
+    if (!utterance)
+        return;
+
+    auto text = utterance->text().utf8();
+    auto spielUtterance = adoptGRef(spiel_utterance_new(text.data()));
+    auto language = utterance->lang().utf8();
+    spiel_utterance_set_language(spielUtterance.get(), language.data());
+    const auto& uri = utterance->voice()->voiceURI();
+    for (auto& voice : m_voices) {
+        auto identifier = StringView::fromLatin1(spiel_voice_get_identifier(voice.get()));
+        if (uri == identifier) {
+            spiel_utterance_set_voice(spielUtterance.get(), voice.get());
+            break;
+        }
+    }
+    spiel_utterance_set_volume(spielUtterance.get(), utterance->volume());
+    spiel_utterance_set_pitch(spielUtterance.get(), utterance->pitch());
+    spiel_utterance_set_rate(spielUtterance.get(), utterance->rate());
```

Comment:
Maybe not in an initial patch, but you can try to detect if the string is XML, and if it is call `spiel_utterance_set_ssml`